### PR TITLE
 Deploy-CmAzProject - First Implementation

### LINKED ISSUE #257: Deploy-CmAzProject
We have a directory deployment tool used by the engineering team that we'd like to incorporate into Cloudmarque. This loops through directories in a structured way and is capable of deploying:

1. Cloudmarque YAML files
2. ARM templates (with parameters file)
3. Bicep
4. PowerShell files
5. Directories containing the above

We should add some unit tests where possible to ensure that this runs smoothly, and document any command options.

Unit tests to include:

* Supported file types.
* Directory targeting.
* Deployment order according to file/directory name
* Incorporate WhatIf testing for deployments.
* Investigate potential unit testing routes for Bicep.

### PR DESCRIPTION
**Pull request by deepend-dev**
closes #257

### CODE REVIEW

File: Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1 [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+         Deploy-CmAzProject -SettingsFile "c:/directory/settingsFile.yml"
+
+        .Example
+		 Deploy-CmAzProject -SettingsFile "c:/directory/settingsFile.yml" -TagSettingsFile "c:/directory/tags.yml"
+
+        .Example
+		 Deploy-CmAzProject -SettingsFile "c:/directory/virtualmachine.yml" -LocalAdminUsername $adminUser -LocalAdminPassword $adminPassword
+	#>
+
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Medium")]
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1:37]: ConfirmImpact = "High"

File: Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1 [hunk 2]
```diff
@@ -0,0 +1,63 @@
+function Deploy-CmAzProject {
+
+    <#
+		.Synopsis
+		 Auto-deploys Azure resources depending on the type of component defined in settings.
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1:5]: Auto deploys resources specified in targeted directories and files of various types.

File: Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1 [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+        .Description
+		 Wrapper function to auto-identify background script to process provided settings and deploy Azure infrastructure.
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1:8]: Auto deploys resources with the following file types:
* Powershell (.ps1)
* Arm Templates (.json)
* Azure Bicep file (.bicep)
* Cloudmarque Yml files via New-CmAzDeployment
* Directories containing any of the above.

Deployment order is automatically determined via the numeric prefix of the directory/file names.

File: Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1 [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        .Parameter SettingsFile
+		 File path for the settings file to be converted into a settings object.
+
+        .Parameter SettingsObject
+         Object containing the configuration values required to run this cmdlet.
+
+        .Parameter TagSettingsFile
+         File path for settings containing tags definition.
+
+			* Uppercase
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1:19]: Remove this

File: Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1 [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+			* Special
+
+        .Component
+		 Common
+
+        .Example
+         Deploy-CmAzProject -SettingsFile "c:/directory/settingsFile.yml"
+
+        .Example
+		 Deploy-CmAzProject -SettingsFile "c:/directory/settingsFile.yml" -TagSettingsFile "c:/directory/tags.yml"
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1:31]: No need for tag settings file i think

File: Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1 [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+	#>
+
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Medium")]
+
+    param(
+        [parameter(Mandatory = $true, ParameterSetName = "Settings File")]
+        [String]$SettingsFile,
+        [parameter(Mandatory = $true, ParameterSetName = "Settings Object")]
+        [Object]$SettingsObject,
+        [String]$TagSettingsFile
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1:44]: No need for tag settings file i think

File: Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1 (hunk 1 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Deploy-CmAzProject.ps1:28]: done

**Pull request closed without merging**